tools/run-tests.py

```diff
@@ -331,56 +340,34 @@ def run_buildoption_test(options):
 
     return ret
 
-def main(options):
-    ret = 0
-
-    if options.check_signed_off_tolerant:
-        ret = run_check([settings.SIGNED_OFF_SCRIPT, '--tolerant'])
-
-    if not ret and options.check_signed_off_travis:
-        ret = run_check([settings.SIGNED_OFF_SCRIPT, '--travis'])
-
-    if not ret and (options.all or options.check_signed_off):
-        ret = run_check([settings.SIGNED_OFF_SCRIPT])
-
-    if not ret and (options.all or options.check_cppcheck):
-        ret = run_check([settings.CPPCHECK_SCRIPT])
-
-    if not ret and (options.all or options.check_doxygen):
-        ret = run_check([settings.DOXYGEN_SCRIPT])
-
-    if not ret and (options.all or options.check_pylint):
-        ret = run_check([settings.PYLINT_SCRIPT])
-
-    if not ret and (options.all or options.check_vera):
-        ret = run_check([settings.VERA_SCRIPT])
-
-    if not ret and (options.all or options.check_license):
-        ret = run_check([settings.LICENSE_SCRIPT])
-
-    if not ret and (options.all or options.check_magic_strings):
-        ret = run_check([settings.MAGIC_STRINGS_SCRIPT])
-
-    if not ret and (options.all or options.jerry_debugger):
-        ret = run_jerry_debugger_tests(options)
-
-    if not ret and (options.all or options.jerry_tests):
-        ret = run_jerry_tests(options)
-
-    if not ret and (options.all or options.jerry_test_suite):
-        ret = run_jerry_test_suite(options)
-
-    if not ret and (options.all or options.test262):
-        ret = run_test262_test_suite(options)
-
-    if not ret and (options.all or options.unittests):
-        ret = run_unittests(options)
-
-    if not ret and (options.all or options.buildoption_test):
-        ret = run_buildoption_test(options)
-
-    sys.exit(ret)
+class Check(object):
```

Comment:
For such cases there are namedtuples: https://docs.python.org/2/library/collections.html#collections.namedtuple

Reply:
BTW, namedtuples don't allow `*args`, do they? (making classes more powerful/future-proof)

Reply:
It would be great to use namedtuple here (and also at the Options, again another PR imho). The problem currently is that the pylint returns a warning: `Too few public methods (0/2) (too-few-public-methods)` which of course we could change.

What if we do this?:
```
_Check = namedtuple("Check", ["enabled", "runner", "args"])
def Check(enabled, runner, *args):
    return _Check(enabled, runner, args)
```
With this we have support for the `*args` also.

Reply:
Honestly, the only thing the pylint warning signals to me is that we don't have pylint configured properly. I think pylint's own documentation is quite open about this:

> What Pylint says is not to be taken as gospel and Pylint isn’t smarter than you are: it may warn you about things that you have conscientiously done. Pylint tries hard to report as few false positives as possible for errors, but it may be too verbose with warnings.

In another project, I've been experimenting with reasonable settings for pylint and this was that worked for us: https://github.com/renatahodovan/grammarinator/blob/master/.pylintrc (It's quite smaller and also more maintainable than what we have now in JerryScript.)

On `Check` and `_Check`: for the current purposes, the suggested approach is functionally equivalent to the PR's approach but less pythonic and/or less OO: values returned by `Check` are not instances of `Check`.

Reply:
I've pushed an experimental commit on top of the original showing how a (IMO) better maintained pylintrc would work. Please have a look.